utils/merge_syft_sbom.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+import json
+from argparse import ArgumentParser
+from typing import Any, Callable
+from urllib.parse import quote_plus, urlsplit
+
+
+def _is_syft_local_golang_component(component: dict) -> bool:
+    """
+    Check if a Syft Golang reported component is a local replacement.
+
+    Local replacements are reported in a very different way by Cachi2, which is why the same
+    reports by Syft should be removed.
+    """
+    return component.get("purl", "").startswith("pkg:golang") and (
+        component.get("name", "").startswith(".") or component.get("version", "") == "(devel)"
+    )
+
+
+def _is_syft_non_registry_npm_component(component: dict) -> bool:
+    """
+    Check if a Syft NPM component was fetched from a VCS or HTTPS location.
+
+    Cachi2 reports non-registry NPM components in a different way from Syft, so the reports
+    from Syft need to be removed.
+    """
+    version = component.get("version", "")
+    return component.get("purl", "").startswith("pkg:npm") and (
+        version.startswith("https://")
+        or version.startswith("git+https://")
+        or version.startswith("github:")
```

Comment:
I guess that `bitbucket:` and `gitlab:` might also be valid in purls, but I still couldn't get a Syft SBOM with these type of dependencies in it.

Reply:
Maybe we just check if the version is any url by checking if there's a `:`? (Same logic as https://github.com/containerbuildsystem/cachi2/pull/229#discussion_r1259464017)

Reply:
> Maybe we just check if the version is any url by checking if there's a :
Yeah, that might do it :)

Reply:
I'll double check the behavior for v3 lockfiles.

Reply:
Syft does use only the version for lockfiles v2/v3 :facepalm: 

So this means we need to either:
- fallback to the name-only deduping method we are using for Pip
- live with the duplicates
- update Cachi2 so it also outputs the version for file/vcs dependencies